### PR TITLE
refactor(client): remove hasNoTests by challenge type

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This is one step in moving away from our reliance/use of `challengeTypes`.

### CODE REVIEW

File: client/src/templates/Challenges/redux/completion-epic.js [hunk 1]
```diff
@@ -100,10 +99,7 @@ function postChallenge(update, username) {
 function submitModern(type, state) {
   const challengeType = state.challenge.challengeMeta.challengeType;
   const tests = challengeTestsSelector(state);
-  if (
-    hasNoTests(challengeType) ||
-    (tests.length > 0 && tests.every(test => test.pass && !test.err))
-  ) {
+  if (tests.every(test => test.pass && !test.err)) {
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/redux/completion-epic.js:102]: This will return `true` when `tests` is an empty array - is that desired?

**Review by naomi-lgbt — APPROVED** [review]
I'd rather be explicit, but I suppose this works.

File: client/src/templates/Challenges/redux/completion-epic.js (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/redux/completion-epic.js:102]: It's the same, but it's more declarative.

`if (tests.length > 0 && tests.every...)` reads like I would think "if there are tests and they all pass, then..."

`if(test.every...)` makes "if there are tests" implicit. It's not a big deal, but that's the reasoning.

> This is not a feature.

Sorry, yeah, that's true. There probably should be one, though. I don't think it's good that we can forget to add tests and have that challenge default to being submittable.

Anyways, that's a problem for another day.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

Just for peace of mind, could you add `.required()` to the tests property in the challenge schema?

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Looks like snapshot needs an update

**Comment by Nirajn2311** [comment]
The PR's good to go from mobile perspective.

**Review by naomi-lgbt — APPROVED** [review]

**Comment by ShaunSHamilton** [comment]
Thank you for taking care of that snapshot, @naomi-lgbt ⭐

**Pull request merged**